### PR TITLE
Magic

### PR DESCRIPTION
**Pull request by scobb**

### CODE REVIEW

**Comment by ritwik20** [comment]
Looks like samples/magic/__init__.py has no changes?

File: samples/magic/az_ml_magic.py [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+        sys.stdout = self._stdout
+
+
+@magics_class
+class AMLHelpers(Magics):
+    @staticmethod
+    def print_and_update_env(k, v):
+        os.environ[k] = v
+        print(' os.environ["{}"]="{}"'.format(k, v))
+        return 'export {}={}'.format(k, v)
```
Comment (ritwik20) [inline samples/magic/az_ml_magic.py]: So this export statement will only work in a bash-like shell, right? Can we do something like what the CLI does to print these statements based on the OS?

File: samples/magic/az_ml_magic.py [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+
+    @line_magic
+    def list_subs(self, line):
+        from azure.cli.core._profile import Profile
+        try:
+            from azure.cli.core.util import CLIError
+        except ImportError:
+            from azure.cli.core._util import CLIError
+        self._redirect_logging('az.azure.cli.core._profile')
+        profile = Profile()
```
Comment (ritwik20) [inline samples/magic/az_ml_magic.py]: Is this what prompts for login? Can you add a comment describing what Profile() does?

File: samples/magic/az_ml_magic.py [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+    def aml_env_setup(self, line):
+        from azure.cli.core._profile import Profile
+        self._redirect_logging('az.azure.cli.core._profile')
+        p = argparse.ArgumentParser()
+        p.add_argument('-n', '--name', help='base name for your environment',
+                       required=True)
+        p.add_argument('-k', dest='kubernetes',
+                       help='Flag to indicate kubernetes environment', required=False,
+                       action='store_true')
+        p.add_argument('-l', dest='local_only', help='Flag to preclude ACS deployment',
```
Comment (ritwik20) [inline samples/magic/az_ml_magic.py]: Nit: replace preclude with exclude.

File: samples/magic/az_ml_magic.py [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+                      parsed_args.service_principal_password,
+                      c)
+        acs_regex = r"az ml env setup -s (?P<deployment_id>[^']+)"
+        env_regex = r'export (?P<k>[^=]+)=(?P<v>.+)'
+
+        for line in output:
+            s = re.search(acs_regex, line)
+            if s:
+                print(
+                    'To check the status of the deployment, run line magic %check_deployment -d {}'.format(
```
Comment (ritwik20) [inline samples/magic/az_ml_magic.py]: Should this be updated in light of the deployment id no longer being required?

File: samples/magic/az_ml_magic.py [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+        context.local_mode = True
+        context.set_input_response(
+            'Delete existing service and create new service (y/N)? ',
+            'y' if args.overwrite else 'n')
+        _, fp = tempfile.mkstemp()
+        with open(fp, 'w') as score_file:
+            score_file.write(cell)
+        try:
+            print('filepath: {}'.format(fp))
+            print('args.schema: {}'.format(args.schema))
```
Comment (ritwik20) [inline samples/magic/az_ml_magic.py]: Why print just the schema file and the driver file?

File: samples/magic/az_ml_magic.py [hunk 6]
```diff
@@ -0,0 +1,300 @@
...
+                       default=1, required=False, help='Number of replicas for a Kubernetes service.')
+        args = p.parse_args(parameter_s.split())
+        context = u.JupyterContext()
+        context.local_mode = True
+        context.set_input_response(
+            'Delete existing service and create new service (y/N)? ',
+            'y' if args.overwrite else 'n')
+        _, fp = tempfile.mkstemp()
+        with open(fp, 'w') as score_file:
+            score_file.write(cell)
```
Comment (ritwik20) [inline samples/magic/az_ml_magic.py:200]: So this doesn't write the magic invocation line at the top of the cell itself?

**Review by ritwik20 — DISMISSED** [review]

File: samples/magic/az_ml_magic.py (hunk 1 above)
Comment (scobb) [reply samples/magic/az_ml_magic.py]: Yes. I was making an assumption that the machine running the notebook would be Linux. Can fix.

File: samples/magic/az_ml_magic.py (hunk 2 above)
Comment (scobb) [reply samples/magic/az_ml_magic.py]: Sure.

File: samples/magic/az_ml_magic.py (hunk 6 above)
Comment (scobb) [reply samples/magic/az_ml_magic.py:200]: No, it does not.

File: samples/magic/az_ml_magic.py (hunk 5 above)
Comment (scobb) [reply samples/magic/az_ml_magic.py]: Debug print statements. Removing.

**Comment by scobb** [comment]
__init__.py is an added file. Not sure why it shows as "no change."

**Review by ritwik20 — APPROVED** [review]

**Pull request merged**